File: grida-domains/domain/src/main/java/org/grida/domain/user/ProfileImageRepository.java
package org.grida.domain.user;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

public interface ProfileImageRepository {
    long save(String userEmail, ImageDetail imageDetail, LocalDateTime lastActionAt);
    ProfileImage findById(long userEmail);
    String findActivateImagePathByUserEmail(String userEmail);
    List<ProfileImage> findAllImagesByUserEmail(String userEmail);
    LocalDate findMostRecentCreatedDateByUserEmail(String userEmail);
    boolean hasActivateImage(String userEmail);
    void deactivateImage(String userEmail, LocalDateTime lastActionAt);
    void activateImage(long id, LocalDateTime lastActionAt);
}

File: grida-core/core-domain/src/main/kotlin/org/grida/domain/profileimage/ProfileImageRepository.kt
package org.grida.domain.profileimage

import org.grida.domain.image.ImageStatus
import org.grida.domain.user.User

interface ProfileImageRepository {

    fun save(profileImage: ProfileImage, user: User): Long

    fun findById(id: Long): ProfileImage

    fun findByUserIdAndStatus(userId: Long, status: ImageStatus): ProfileImage

    fun findAllByUserId(userId: Long): List<ProfileImage>

    fun existsByUserIdAndStatus(userId: Long, status: ImageStatus): Boolean

    fun updateStatue(profileImageId: Long, status: ImageStatus): Long
}
